src/chunker.c

```diff
@@ -8,13 +8,70 @@
  * optimization. See RFC1950 for details. */
 #define BYTES_PUSH_MAX 5552U
 
+/* Unless otherwise specified, take this as the default average chunk size */
+#define DEFAULT_CHUNK_SIZE_AVG (16 * 1024)
+
+static bool number_is_prime(unsigned int n) {
+        unsigned int i;
+
+        for (i = 2; i < n; i++)
+                if (n % i == 0)
+                        return false;
+
+        return true;
+}
+
+int ca_chunker_set_avg_size(CaChunker *c, size_t avg) {
+        size_t delta, closest;
+
+        assert(c);
+
+        /* Find an average block size that is close to the requested number but is prime,
+         * because this number is used in the modulo operation that determines the chunk
+         * break points and we don't want to waste any bits there. */
+
+        if (c->a > 1)
+                return -EBUSY;
+
+        for (delta = 0; ; delta++) {
+                if (delta > SIZE_MAX / 2)
+                        return -EINVAL;
+
+                closest = avg - delta;
+                if (closest < 3)
+                        return -EINVAL;
+
+                if (number_is_prime(closest))
+                        break;
+
+                closest = avg + delta;
+                if (number_is_prime(closest))
+                        break;
+        }
+
+        c->chunk_size_avg = closest;
+        c->chunk_size_min = (c->chunk_size_avg >> 1) & ~0xff;
+        c->chunk_size_max = c->chunk_size_min << 2;
+
+        if (!c->chunk_size_min)
```

Comment:
i prefer numeric comparisons with == 0, instead of relying on C's downgrade-to-bool feature...